File: app/models/project.rb
class Project
  include Mongoid::Document
  include Mongoid::Timestamps
  include Mongoid::Slug

  field :key, type: String
  field :financing_source, type: String
  field :name, type: String
  field :agreement, type: String
  field :sponsor, type: String
  field :start_date, type: Date
  field :end_date, type: Date
  field :inprogress, type: Boolean
  field :authorized_budget, type: Float
  field :deposited_budget, type: Float
  field :committed_budget, type: Float
  field :spent_budget, type: Float

  embeds_many :lines
  slug :name
end

File: my-frontend/app/models/project.js
import DS from 'ember-data';

export default DS.Model.extend({
  key: DS.attr('string'),
  financing_source: DS.attr('string'),
  name: DS.attr('string'),
  sponsor: DS.attr('string'),
  start_date: DS.attr('date'),
  end_date: DS.attr('date'),
  authorized_budget: DS.attr('string'),
  deposited_budget: DS.attr('string'),
  committed_budget: DS.attr('string'),
  spent_budget: DS.attr('string'),

  lines: DS.hasMany('line', {async: true})
});
